src/charm.py

```diff
@@ -72,11 +74,20 @@ def __init__(self, *args):
         """Initialize defaults and event handlers."""
         super().__init__(*args)
 
+        self._database = DatabaseObserver(self)
+
+        self.framework.observe(
+            self._database.database.on.database_created, self._database_relation_changed
```

Comment:
Is this using the observer pattern for database? If so, what is the reason for putting the event handlers here? Also, the name of the event handling functions should match the name of the event

Reply:
Also applicable to the other event handlers being added

Reply:
This is not using the observer pattern. I propose to revisit this at a later date because using the observer pattern here introduces a fair amount of code and the primary goal was to [deploy discourse in production ASAP](https://warthogs.atlassian.net/browse/ISD-481).  
I've already created [a story for that](https://warthogs.atlassian.net/browse/ISD-934) and it will be my priority after the deployments.